src/main/java/de/jtheb/fh/lagerverwaltung/entities/Shelf.java

```diff
@@ -6,24 +6,26 @@
 public class Shelf {
     public static final int COMPARTMENTCOUNT = 100;
 
-    private List<Compartment> compartments;
+    private List<Compartment> compartments = new ArrayList<>();
 
     public Shelf() {
         this.compartments = new ArrayList<>();
     }
 
     public boolean itemFits(Item item) {
-        for (Compartment compartment : this.compartments) {
-            if (compartment.itemFits(item)) {
-                return true;
+        if (!isFull()) {
```

Comment:
as stated above, isFull will super rarely be true. it won't hurt to iterate over all compartments and return false at the end like we do now